src/cache_cf.cc

```diff
@@ -1121,8 +1121,8 @@ CheckTimeValue(const double value, const std::chrono::nanoseconds &unit)
         throw TexcHere("time must have a positive value");
 
     const auto maxNanoseconds = std::chrono::nanoseconds::max().count();
```

Comment:
It looks like you want to cast all maxNanoseconds to double. If so, here is a better way to do it (replacing your changes -- I cannot undo them in a suggestion):

```suggestion
    const auto maxNanoseconds = static_cast<double>(std::chrono::nanoseconds::max().count());
```

FWIW, I suspect that we are doing this whole check/conversion wrong. There is probably an std::chrono-correct way to do it without so many mysterious (and apparently failing) casts. However, I do not have the time to research a better approach, and it is arguably outside this build-fixing PR scope anyway.

Reply:
There is std::chrono::duration_cast which appears to be designed for this usage.

Reply:
Moved to that, also clarifying intent and removing approximations.